test/controllers/auctions_controller_test.rb

```diff
@@ -0,0 +1,53 @@
+require 'test_helper'
+require 'date'
+
+class AuctionsControllerTest < ActionDispatch::IntegrationTest
+  test '/auctions should retrieve all auctions as json' do
+    auctions_to_create = [
+      {
+        title: 'Mustang',
+        description: '1968 Ford Mustang',
+        starting_price: 1_200_000,
+        ends_at: 10.days.from_now,
+      },
+      {
+        title: 'Fallen Empires Booster',
+        description: '3x MTG Fallen Empires booster packs',
+        starting_price: 1_000,
+        ends_at: 10.days.from_now,
+      }
+    ]
+
+    auctions_to_create.each { |auction| Auction.create(auction) }
+
+    get '/auctions'
+    auction_response = JSON.parse(@response.body, symbolize_names: true)
+    response_auction_titles = auction_response.map {|auction| auction[:title]}
+
+    assert_response :success
+    assert_equal auctions_to_create.count, auction_response.count
+    assert_equal auctions_to_create.map {|auction| auction[:title]}.sort, response_auction_titles.sort
+  end
+
+  test 'should return the correct auction given an id' do
+    auction_to_create = {
+      title: 'Mustang',
+      description: '1968 Ford Mustang',
+      starting_price: 1_200_000,
+      ends_at: 10.days.from_now.strftime('%FT%TZ')
+    }
+    created_auction = Auction.create(auction_to_create)
+
+    get "/auctions/#{created_auction.id}"
+    api_response = JSON.parse(@response.body, symbolize_names: true)
+
+    assert_response :success
```

Comment:
You don't have to change it in this PR, but the idea of this check is to have it before the JSON parsing.
Otherwise, let's say you're getting a 401 or 500 with an empty body, the `JSON.parse` would throw a weird "can't parse JSON" error that it's difficult to understand.

What I usually do is create a test helper that checks for the status and parses the body at the same time:
Something like `parse_json_body(expected_status: 401)`, and you can default `expected_status` to 200 (you could use the symbols as well).